MTA-6893 Document Hub OIDC authentication (#382)
migtools/mta-documentation: docs/topics/mta-install/proc_create-custom-role-assign-permissions.adoc

:_newdoc-version: 2.18.5
:_template-generated: 2025-07-31
:_mod-docs-content-type: PROCEDURE

[id="create-custom-role-assign-permissions_{context}"]
= Create a custom role and assign permissions

[role="_abstract"]

> To enforce fine-grained role-based access control (RBAC) permissions for users to perform specific tasks, create custom roles and assign specific permissions in the {ProductFullName} web console. {ProductShortName} exposes features based on the scope of the permissions you assign.

Reviewer: ```suggestion
To enforce fine-grained role-based access control (RBAC), create custom roles and assign specific permissions in the {ProductFullName} web console. Users can access features based on the permissions that you assign to them.
```

User Focus and Personas: shifted the perspective from third-person generic personas (e.g., "Administrators can...") to the active second-person (e.g., "As an administrator, you can...") to create a more engaging, conversational tone. 

Lowercased generic personas like "administrator", "architect", and "migrator" when they appeared in the middle of sentences.

Minimalism and Fluff: removed filler phrases (e.g., "If necessary", "information related to the applications") and wordy transitions (e.g., "However, the Migrator cannot"). We also replaced industry jargon like "consume" or colloquialisms like "gets" with simpler, more precise verbs like "use", "view", or "receive".

Anthropomorphism: corrected sentences that assigned human actions to software. For example, instead of saying "{ProductShortName} installs" or "{ProductShortName} exposes," we attributed the actions to the Operator ("the Operator installs") or the user ("Users can access features").

Pronouns and Articles: added definite and indefinite articles ("the", "a", "an") and the word "that" before dependent clauses to improve clarity and machine translation. 

Used pronouns like "they" and "them" to avoid repeating the same nouns within a paragraph

[NOTE] 
====
You cannot alter or delete the default roles and permissions scoped to them.
====

.Prerequisites

* You are an {ProductShortName} Administrator user.
* You installed {ProductShortName} 8.2.0.

.Procedure

> . Log in to the {ProductShortName} web console.
> . In the Administration mode, go to *User Management*.
> . Click *Roles*.
> . Click the number of permissions scoped to the `admin` role to explore the resources and scoped permissions.
> . On the Roles page, click *Create*.
> . Enter a unique name for the new role in the *Name* field.
> . Type the name of a resource to see the list of permissions.
> . Add one or more permissions:
> * Select a permission from the *Permissions* list and click the *Add selected* icon.
> * Click the *Add all* icon to select all the permissions for that resource.
> . Click *Create*.

Reviewer: ```suggestion
. Log in to the {ProductShortName} web console.
. In the *Administration* mode, click *User Management*.
. Click *Roles*.
. Click the number of permissions scoped to the `admin` role to explore the resources and scoped permissions.
. On the *Roles* page, click *Create*.
. In the *Name* field, enter a unique name for the new role.
. Type the name of a resource to see the list of permissions.
. Add one or more permissions by using one of the following methods:
* From the *Permissions* list, select a permission and click the *Add selected* icon.
* Click the *Add all* icon to select all the permissions for that resource.
. Click *Create*.
```

Shift from third-person generic personas (e.g., "Administrators can...") to the active second-person (e.g., "As an administrator, you can...") to maintain an engaging, conversational tone.

Avoid assigning human actions to software. Instead of saying the product itself "installs," "allows," or "exposes" features, attribute the actions to the Operator ("the Operator installs") or the user ("Users can access features").

Include definite/indefinite articles ("the", "a") and the relative pronoun "that" before dependent clauses to improve clarity and machine translation.

Adhere to official product names, such as using "Red Hat build of Keycloak" (with a lowercase 'b') and expanding "OpenShift" to "Red Hat OpenShift Container Platform" on its first occurrence. 

Write prerequisite statements in the present perfect tense (e.g., "You have installed...") to represent completed actions and maintain grammatical parallelism across standard lists.

Location First: In procedural steps, provide locational context before the action (e.g., "On the Edit user page, specify...") so users know where to look before they execute a task.

All UI elements. Replace vague navigation verbs like "go to" with precise actions like "click" or "select", and replace terms like "dropdown" with "drop-down list" or "menu".
